src/LaunchDarkly.ServerSdk/Internal/Model/Clause.cs

```diff
@@ -63,19 +63,32 @@ private static PreprocessedData Preprocess(Operator op, IEnumerable<LdValue> val
                         return new PreprocessedValue();
                     }
                 }).ToImmutableList()
-            };
+            );
 
-        internal struct PreprocessedData
+        internal readonly struct PreprocessedData
         {
-            internal ImmutableHashSet<LdValue> ValuesAsSet { get; set; }
-            internal ImmutableList<PreprocessedValue> Values { get; set; }
+            internal ImmutableHashSet<LdValue> ValuesAsSet { get; }
+            internal ImmutableList<PreprocessedValue> Values { get; }
+
+            public PreprocessedData(ImmutableHashSet<LdValue> valuesAsSet = null, ImmutableList<PreprocessedValue> values = null)
```

Comment:
It's either or so I just matched the equivalent getter/setter interface that existed